src/main/java/org/folio/linked/data/validation/dto/PrimaryTitleDtoValidator.java

```diff
@@ -0,0 +1,28 @@
+package org.folio.linked.data.validation.dto;
+
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.linked.data.domain.dto.PrimaryTitleField;
+import org.folio.linked.data.domain.dto.TitleField;
+import org.folio.linked.data.validation.PrimaryTitleConstraint;
+
+public class PrimaryTitleDtoValidator implements ConstraintValidator<PrimaryTitleConstraint, List<TitleField>> {
+
+  @Override
+  public boolean isValid(List<TitleField> titleFields, ConstraintValidatorContext context) {
+    return ofNullable(titleFields)
+      .map(tfs -> tfs.stream()
+        .filter(PrimaryTitleField.class::isInstance)
+        .map(ptf -> ((PrimaryTitleField) ptf).getPrimaryTitle())
+        .filter(pt -> isNotEmpty(pt.getMainTitle()))
+        .flatMap(pt -> pt.getMainTitle().stream())
+        .anyMatch(StringUtils::isNotBlank))
+      .orElse(true);
```

Comment:
minor
discussion about complicated map
`if` in this case looks better, just because you can see return at the start of the method and no complicated map functions

Reply:
But also since we have @NotNull validation check at the same field, we could skip a null check here at all, so removed it

Reply:
We can't remove it. As I know validators wor independently and combine problems from all operators.
It means that we have NullPointerException in case it's null and crash validation at all

Reply:
Actually, no, combination of @NotNull + 2nd constraint works well without null-check in 2nd constraint.
See by yourself: org.folio.linked.data.e2e.ResourceControllerValidationIT#createEmptyInstance_shouldReturnBadRequest

Reply:
But it doesn't feel safe.. ok, returned a null check